Content/Items/Misc/Accessories.TaintedGreataxe.cs

```diff
@@ -0,0 +1,478 @@
+    class TaintedGreataxeProjectile : ModProjectile
+    {
+        public bool init = false;
+        private List<Vector2> oldPosition = new List<Vector2>();
+        private List<float> oldRotation = new List<float>();
+
+        public NPC targetNPC;
+        public bool Embedding;
+        public int EmbeddingCooldown;
+
+        public bool stickyAI;
+        public Vector2 offset = Vector2.Zero;
+        public int enemyWhoAmI; 
+
+        public Player Owner => Main.player[Projectile.owner];
+
+        public override string Texture => AssetDirectory.MiscItem + Name;
+
+        public override void SetStaticDefaults()
+        {
+			DisplayName.SetDefault("Ethereal Greataxe");
+            ProjectileID.Sets.TrailingMode[Projectile.type] = 2;
+            ProjectileID.Sets.TrailCacheLength[Projectile.type] = 12;
+        }
+
+        public override void SetDefaults()
+        {
+            Projectile.friendly = true;
+			Projectile.DamageType = DamageClass.Generic;
+
+			Projectile.tileCollide = false;
+			Projectile.penetrate = -1;
+
+            Projectile.width = 14;
+            Projectile.height = 14;
+
+            Projectile.usesLocalNPCImmunity = true;
+            Projectile.localNPCHitCooldown = 20;
+        }
+
+        public override bool PreAI()
+        {
+            if (stickyAI)
+            {
+                NPC target = Main.npc[enemyWhoAmI];
+                Projectile.position = target.position + offset;
+            }
+            return true;
+        }
+
+        public override void AI()
+        {
+            if (!init)
+            {
+                init = true;
+                oldPosition = new List<Vector2>();
+                oldRotation = new List<float>();
+            }
+
+            if (Embedding && EmbeddingCooldown <= 0)
+                EmbeddingAI();           
+            else
+                IdleMovement();
+
+            if (Owner.dead)
+                Projectile.Kill();
+
+            oldRotation.Add(Projectile.rotation);
+            oldPosition.Add(Projectile.Center);
+
+            while (oldRotation.Count > 12)
+                oldRotation.RemoveAt(0);
+            while (oldPosition.Count > 12)
+                oldPosition.RemoveAt(0);
+
+            if (EmbeddingCooldown > 0)
+                EmbeddingCooldown--;
+        }
+
+        public override bool PreDraw(ref Color lightColor)
+        {
+            Texture2D tex = ModContent.Request<Texture2D>(Texture).Value;
+            SpriteEffects spriteEffects = SpriteEffects.None;
+            if (Embedding)
+                spriteEffects = Projectile.direction == -1 ? SpriteEffects.FlipHorizontally : SpriteEffects.None;
+
+            Vector2 origin = tex.Size() / 2f;
+            origin.Y = (Projectile.spriteDirection == 1 ? tex.Height - 40 : 40);
+
+            Main.spriteBatch.End();
+            Main.spriteBatch.Begin(default, BlendState.Additive, default, default, default, default, Main.GameViewMatrix.TransformationMatrix);
```

Comment:
Do we need to reset the blendstate here or can we just set the alpha channel on the color to 0?